src/js/actions/documents.js

```diff
@@ -647,11 +665,14 @@ define(function (require, exports) {
                 return descriptor.playObject(documentLib.select(documentRef));
             })
             .then(function () {
-                var payload = {
-                    selectedDocumentID: document.id
-                };
-
-                this.dispatch(events.document.SELECT_DOCUMENT, payload);
+                if (!document.layers) {
+                    // The now-active document has yet to be fully initialized.
+                    return this.transfer(updateDocument);
+                } else {
+                    this.dispatch(events.document.SELECT_DOCUMENT, {
```

Comment:
did you intend not to use `return this.dispatchAsync...`?  Also, in the case that you transfer to updateDocument, wouldn't we also need to emit SELECT_DOCUMENT for the application store?